File: java/02-internals/users/src/main/java/com/apress/users/UsersController.java
package com.apress.users;

import org.springframework.web.bind.annotation.*;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/users")
public class UsersController {
    private Map<String,User> users = new HashMap<>() {{
        put("ximena@email.com",new User("ximena@email.com","Ximena"));
        put("norma@email.com",new User("norma@email.com","Norma"));

    }};

    @GetMapping
    public Collection<User> getAll(){
        return this.users.values();
    }

    @GetMapping("/{email}")
    public User findUserByEmail(@PathVariable String email){
        return this.users.get(email);
    }

    @PostMapping
    public User save(@RequestBody User user){
        this.users.put(user.getEmail(),user);
        return user;
    }

    @DeleteMapping("/{email}")
    public void save(@PathVariable String email){
        this.users.remove(email);
    }
}

File: kotlin/01-quick-start/users/src/main/kotlin/com/apress/users/UsersController.kt
package com.apress.users

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UsersController {
    private val users: HashMap<String?, User?> = object : HashMap<String?, User?>() {
        init {
            put("ximena@email.com", User("ximena@email.com", "Ximena"))
            put("norma@email.com", User("norma@email.com", "Norma"))
        }
    }

    @GetMapping
    fun getAll(): MutableCollection<User?> = users.values

    @GetMapping("/{email}")
    fun findUserByEmail(@PathVariable email: String?): User? {
        return users[email]
    }

    @PostMapping
    fun save(@RequestBody user: User): User {
        users[user.email] = user
        return user
    }

    @DeleteMapping("/{email}")
    fun save(@PathVariable email: String?) {
        users.remove(email)
    }
}